Mathlib/Analysis/InnerProductSpace/Symmetric.lean

```diff
@@ -74,17 +74,48 @@ theorem IsSymmetric.apply_clm {T : E →L[𝕜] E} (hT : IsSymmetric (T : E →
     ⟪T x, y⟫ = ⟪x, T y⟫ :=
   hT x y
 
+@[aesop safe apply]
 theorem isSymmetric_zero : (0 : E →ₗ[𝕜] E).IsSymmetric := fun x y =>
   (inner_zero_right x : ⟪x, 0⟫ = 0).symm ▸ (inner_zero_left y : ⟪0, y⟫ = 0)
 
+@[aesop safe apply]
 theorem isSymmetric_id : (LinearMap.id : E →ₗ[𝕜] E).IsSymmetric := fun _ _ => rfl
 
-theorem IsSymmetric.add {T S : E →ₗ[𝕜] E} (hT : T.IsSymmetric) (hS : S.IsSymmetric) :
+@[aesop safe apply]
+theorem isSymmetric.add {T S : E →ₗ[𝕜] E} (hT : T.IsSymmetric) (hS : S.IsSymmetric) :
     (T + S).IsSymmetric := by
   intro x y
   rw [LinearMap.add_apply, inner_add_left, hT x y, hS x y, ← inner_add_right]
   rfl
 
+@[aesop safe apply]
+theorem isSymmetric.sub {T S : E →ₗ[𝕜] E} (hT : T.IsSymmetric) (hS : S.IsSymmetric) :
+    (T - S).IsSymmetric := by
+  intro x y
+  rw [LinearMap.sub_apply, inner_sub_left, hT x y, hS x y, ← inner_sub_right]
+  rfl
+
+@[aesop safe apply]
+theorem isSymmetric.smul {T : E →ₗ[𝕜] E} (hT : T.IsSymmetric) {c : 𝕜} (hc : conj c = c) :
+    c • T |>.IsSymmetric := by
+  intro x y
+  simp only [smul_apply, inner_smul_left, hc, hT x y, inner_smul_right]
+
+@[aesop safe apply]
```

Comment:
I certainly don't think this should be marked `safe`, if we should mark it `aesop` at all. The 30% is just a guess. I don't think we have solid guidelines yet for this kind of thing.
```suggestion
@[aesop 30% apply]
```

Reply:
Is this because of the weirdness of trying to resolve the various coercions appearing? I don't really understand `safe` versus `unsafe` beyond that `safe` are tried first, and that `unsafe` have a success probability. (So is the change you are suggesting marking this unsafe, implicitly?)

Reply:
No, it's because `safe` lemmas will never backtrack. `unsafe` lemmas can be backtracked, and the percentage (which *does* imply `unsafe` as you surmised) tells `aesop` "how likely" it is to apply. `aesop` has a priority queue of which lemmas will be applied and the percentage affects where it ends up there.

Think of it this way: if you are going to prove `IsSymmetric (S + T)` (that's not a typo, I do mean `+`), then, unless you're in some weird situation, you almost always want to prove this by showing `S` and `T` are individually symmetric. However, in contrast, if you want to show that `S * T` is symmetric, it's only sometimes the case that you want this to follow from the fact that they are symmetric and commute (and moreover, this generates a `Commute` side condition that has to be discharged as well, so it could be more expensive to try this).

Reply:
Ah!